zanata-war/src/main/java/org/zanata/webtrans/server/TranslationWorkspaceManagerImpl.java

```diff
@@ -172,26 +177,47 @@ protected String getSessionId() {
     }
 
     @Observer(ProjectUpdate.EVENT_NAME)
-    public void projectUpdate(@Observes ProjectUpdate payload) {
-        projectUpdate(payload.getProject());
+    // transaction has already been committed and marked as rolled back only for
+    // current thread. We have to open a new transaction to load any lazy
+    // properties (otherwise exception like javax.resource.ResourceException:
+    // IJ000460: Error checking for a transaction: Transactions are not active)
+    @Async
```

Comment:
It must be in a separate thread to be able to get a new transaction (the reason is in the comment). Because the event is fired in post update commit hibernate listener (which means the transaction in current thread is committed and you can not do anything to get access database again. Unless there is other way to get a new transaction or reuse this committed transaction, this is the only way to let it start a new transaction.